### PR TITLE
Setup tdd

### PR DESCRIPTION
**Pull request by smunoz1988**
Hello!

Team: @danielamoreno699  / @smunoz1988 

For this milestone, we made these changes:

- Create a class called Solver.
- Create a method called factorial that takes one argument, an integer N, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number. The factorial is the multiplication of all integers from 1 to N and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.
- Create a method called reverse that takes one argument, a string word, and returns word reversed (e.g. if word is "hello" it returns "olleh").
- Create a method called fizzbuzz that takes one argument, an integer N, and returns a string. The returned string is constructed following these rules:

    1. When N is divisible by 3, return "fizz".
    2. When N is divisible by 5, return "buzz".
    3. When N is divisible by 3 and 5, return "fizzbuzz".
    4. Any other case, return N as a string (e.g. say N is 7 then return "7").

- As required, We follow TDD practice. We create the tests first and then the methods. 

image

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -9,37 +9,37 @@
...
+            expect(mockInteger).to receive(:factorial).and_return(120)
 
-        allow(Solver).to receive(:new).and_return(mockInteger)
-        solver = Solver.new
-        expect(solver.factorial(5)).to eql(120)
-    end
+            allow(Solver).to receive(:new).and_return(mockInteger)
+            solver = Solver.new
+            expect(solver.factorial(5)).to eql(120)
+        end
```
Comment (abdulhamiid) [inline spec/solver_spec.rb]: Good job! with the passing tests 🤓💯

However, according to the [project requirements](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/testing/tdd_project.md), please add the `#factorial` method test that handles:

- A negative value that raises an exception as an expectation ☺👍

image

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi Team 👋 , 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there! :running_man: :runner: 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🌟

- ✔️ Write unit tests using Rspec.
- ✔️ Understand an existing codebase in a way that allows for adding tests.
- ✔️ Write Rspec unit tests with mocks.
- ✔️ Understand the TDD concept.

## Required Changes ♻️
_Please kindly check the comments under the review._

I'm unable to add an inline comment to your README, please check the required changes below
[REQUIRED] - Kindly fix the broken link in the `Authors` section, let's avoid broken links in our readme for a professional repository. 🤗

image


## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/solver_spec.rb (hunk 1 above)
Comment (smunoz1988) [reply spec/solver_spec.rb]: @abdulhamiid  and @danielamoreno699  Changes done!

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @smunoz1988 & @danielamoreno699, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] All unit tests are passing with green checks :white_check_mark: 
- [x] Pull request has a descriptive title and summary :white_check_mark: 
- [x] Git flow is used correctly :white_check_mark:   
- [x] Implemented all the changes from the previous reviewer :white_check_mark:  

## Optional suggestions
- [ Optional ] I think it would be better if you fix the broken logo image on your readme file. A broken image link on your readme file makes your repository look unprofessional.

Screenshot from 2023-06-30 00-55-16


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**